rcl/include/rcl/client.h

```diff
@@ -0,0 +1,301 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__CLIENT_H_
+#define RCL__CLIENT_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rosidl_generator_c/service_type_support.h"
+
+#include "rcl/macros.h"
+#include "rcl/node.h"
+#include "rcl/visibility_control.h"
+
+/// Internal rcl client implementation struct.
+struct rcl_client_impl_t;
+
+/// Handle for a rcl client.
+typedef struct rcl_client_t
+{
+  struct rcl_client_impl_t * impl;
+} rcl_client_t;
+
+/// Options available for a rcl client.
+typedef struct rcl_client_options_t
+{
+  /// Middleware quality of service settings for the client.
+  rmw_qos_profile_t qos;
+  /// Custom allocator for the client, used for incidental allocations.
+  /* For default behavior (malloc/free), use: rcl_get_default_allocator() */
+  rcl_allocator_t allocator;
+} rcl_client_options_t;
+
+/// Return a rcl_client_t struct with members set to NULL.
+/* Should be called to get a null rcl_client_t before passing to
+ * rcl_initalize_client().
+ * It's also possible to use calloc() instead of this if the rcl_client is
+ * being allocated on the heap.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_client_t
+rcl_get_zero_initialized_client(void);
+
+
+/// Initialize a rcl client.
+/* After calling this function on a rcl_client_t, it can be used to send requests of the given
+ * type by calling rcl_send_request().
+ * Once a response has been sent by the service, the response is available to the client when
+ * rcl_take_response() is called.
```

Comment:
Is it more correct to say "Once a response has been sent by the service, a response may become available in the future and if so it can be retrieved using rcl_take_response()."? As it is currently worded it sounds like calling `rcl_take_response` is what makes it available, which I think is misleading.